diff --git a/target/linux/ramips/dts/rt5350_unbranded_a5-v11.dts b/target/linux/ramips/dts/rt5350_unbranded_a5-v11.dts
index 5f2bad2..605c889 100644
--- a/target/linux/ramips/dts/rt5350_unbranded_a5-v11.dts
+++ b/target/linux/ramips/dts/rt5350_unbranded_a5-v11.dts
@@ -92,7 +92,7 @@
 			partition@50000 {
 				compatible = "denx,uimage";
 				label = "firmware";
-				reg = <0x50000 0x3b0000>;
+				reg = <0x50000 0xfb0000>;
 			};
 		};
 	};
diff --git a/target/linux/ramips/image/rt305x.mk b/target/linux/ramips/image/rt305x.mk
index 815e03a..52c5ebf 100644
--- a/target/linux/ramips/image/rt305x.mk
+++ b/target/linux/ramips/image/rt305x.mk
@@ -1086,10 +1086,10 @@ TARGET_DEVICES += trendnet_tew-714tru
 
 define Device/unbranded_a5-v11
   SOC := rt5350
-  IMAGE_SIZE := 3776k
+  IMAGE_SIZE := 15872k
   IMAGES += factory.bin
   IMAGE/factory.bin := $$(sysupgrade_bin) | check-size | \
-	poray-header -B A5-V11 -F 4M
+	poray-header -B A5-V11 -F 16M
   DEVICE_VENDOR := Unbranded
   DEVICE_MODEL := A5-V11
   DEVICE_PACKAGES := kmod-usb-ohci kmod-usb2
diff --git a/tools/firmware-utils/patches/010-a5-v11-8M.patch b/tools/firmware-utils/patches/010-a5-v11-8M.patch
new file mode 100644
index 0000000..3171334
--- /dev/null
+++ b/tools/firmware-utils/patches/010-a5-v11-8M.patch
@@ -0,0 +1,29 @@
+--- a/src/mkporayfw.c.orig	2023-01-11 00:19:44.809170612 +0000
++++ b/src/mkporayfw.c	2023-01-11 00:21:42.828547055 +0000
+@@ -133,6 +133,9 @@
+ 		.id		= "8M",
+ 		.fw_max_len	= 0x7c0000,
+ 	}, {
++		.id		= "16M",
++		.fw_max_len	= 0xfc0000,
++	}, {
+ 		/* terminating entry */
+ 	}
+ };
+@@ -144,6 +147,16 @@
+ 		.layout_id      = "4M",
+ 		.key            = KEY_A5_V11,
+         }, {
++		.id             = "A5-V11",
++		.hw_id          = HWID_A5_V11,
++		.layout_id      = "8M",
++		.key            = KEY_A5_V11,
++        }, {
++		.id             = "A5-V11",
++		.hw_id          = HWID_A5_V11,
++		.layout_id      = "16M",
++		.key            = KEY_A5_V11,
++        }, {
+ 		.id		= "MPR-A1",
+ 		.hw_id		= HWID_HAME_MPR_A1_L8,
+ 		.layout_id	= "4M",
diff -Nu a/files.orig/etc/uci-defaults/99-custom b/files/etc/uci-defaults/99-custom
--- a/files.orig/etc/uci-defaults/99-custom	1970-01-01 00:00:00.000000000 +0000
+++ b/files/etc/uci-defaults/99-custom	2023-01-13 15:44:59.053702569 +0000
@@ -0,0 +1,40 @@
+[ "$(uci -q get system.@system[0].zonename)" = "Europe/Kyiv" ] && exit 0
+
+uci -q batch << EOF
+set system.@system[0].zonename='Europe/Kyiv'
+add system led
+set system.@led[-1].sysfs='red:power'
+set system.@led[-1].trigger='timer'
+set system.@led[-1].delayon='100'
+set system.@led[-1].delayoff='5000'
+set luci.main.mediaurlbase='/luci-static/openwrt.org'
+set system.@system[-1].hostname='router'
+set system.@system[-1].zram_size_mb='4'
+
+add system led
+set system.@led[-1].sysfs='blue:system'
+set system.@led[-1].trigger='netdev'
+set system.@led[-1].dev='wlan0'
+set system.@led[-1].mode='rx'
+
+set network.wan=interface
+set network.wan.proto='none'
+set network.wan.proto='dhcp'
+set network.wan.device='eth0'
+
+set network.@device[0].ports='wlan0'
+
+set wireless.radio0.disabled='0'
+del wireless.radio0.htmode
+set wireless.radio0.channel='auto'
+set wireless.radio0.cell_density='0'
+
+set wireless.default_radio0.mode='ap'
+set wireless.default_radio0.ssid='OpenWrt'
+set wireless.default_radio0.encryption='none'
+set wireless.default_radio0.network='lan'
+
+add_list dhcp.wan.ra_flags='none'
+
+commit
+EOF
+
